bcbio/structural/purecn.py

```diff
@@ -240,3 +313,95 @@ def _loh_to_vcf(cur):
+
+def process_intervals(data):
+    """Prepare intervals file"""
+    bed_file = regions.get_sv_bed(data)
+    if not bed_file:
+         bed_file = bedutils.clean_file(dd.get_variant_regions(data), data)
+    if not bed_file:
+        return None
+
+    basename = os.path.splitext(bed_file)[0]
+    ready_file = basename + ".txt"
+    if os.path.exists(ready_file):
+        return ready_file
+    optimized_bed = basename + ".optimized.bed"
+    rscript = utils.Rscript_cmd("r36")
+    interval_file_r = utils.R_package_script("r36", "PureCN", "extdata/IntervalFile.R")
+    ref_file = dd.get_ref_file(data)
+    mappability_resource = dd.get_variation_resources(data)["purecn_mappability"]
+    cmd = [rscript, interval_file_r, "--infile", bed_file,
+          "--fasta", ref_file,
+          "--outfile", ready_file, 
+          "--offtarget",
+          "--genome hg38",
+          "--export", optimized_bed,
+          "--mappability", mappability_resource]
+    try:
+        cmd_line = "export R_LIBS_USER=%s && %s && %s" % (utils.R_sitelib(env = "r36"),
+                                                     utils.get_R_exports(env = "r36"),
+                                                     " ".join([str(x) for x in cmd]))
+        do.run(cmd_line, "PureCN intervals")
+    except subprocess.CalledProcessError as msg:
+        logger.info("PureCN failed to prepare intervals")
+    logger.debug("Saved PureCN interval file into " + ready_file)
+    return ready_file
+
+def get_coverage(data):
+    """Calculate coverage for a sample.bam, account for GC content
+       data is single sample
+    """
+    data = utils.to_single_data(data)
+    bed_file = tz.get_in(["config", "algorithm", "purecn_bed_ready"], data)
+    sample_name = dd.get_sample_name(data)
+    work_dir = _sv_workdir(data)
+    rscript = utils.Rscript_cmd("r36")
+    coverage_r = utils.R_package_script("r36", "PureCN", "extdata/Coverage.R")
+    intervals = tz.get_in(["config", "algorithm", "purecn_bed_ready"], data)
+    # PureCN resolves symlinks and the actual output PureCN coverage file name 
+    # is derived from the end bam not from bam_file
+    bam_file = os.path.realpath(dd.get_align_bam(data))
+    bam_name = os.path.basename(bam_file)
+    (bname, ext) = os.path.splitext(bam_name)
+    result_file = os.path.join(work_dir, bname + "_coverage_loess.txt.gz")
+    if not os.path.exists(result_file):
+        cmd = [rscript, coverage_r,
+               "--outdir", work_dir,
+               "--bam", bam_file,
+               "--intervals", intervals]
+        try:
+            cmd_line = "export R_LIBS_USER=%s && %s && %s" % (utils.R_sitelib(env = "r36"),
+                                                              utils.get_R_exports(env = "r36"),
+                                                              " ".join([str(x) for x in cmd]))
+            do.run(cmd_line, "PureCN coverage")
+        except subprocess.CalledProcessError as msg:
```

Comment:
What happens here? (same as other questions) It looks like the error is, in fact, ignored.